src/command_list.rs

```diff
@@ -56,6 +56,34 @@ impl IndirectArgument {
     // TODO: missing variants
 }
 
+#[repr(transparent)]
+pub struct ResourceBarrier(d3d12::D3D12_RESOURCE_BARRIER);
+
+impl ResourceBarrier {
+    pub fn transition(
+        resource: Resource,
+        subresource: u32,
+        before_states: d3d12::D3D12_RESOURCE_STATES,
```

Comment:
Is there a reason for renaming "StateBefore" to "before_states"?
I think there is value in trying to be unopinionated here and just go with "state_before"

Reply:
thanks, done!